libexec/bats-core/bats

```diff
@@ -46,6 +46,7 @@ HELP_TEXT_HEADER
   -p, --pretty              Shorthand for "--formatter pretty"
   -r, --recursive           Include tests in subdirectories
   -t, --tap                 Shorthand for "--formatter tap"
+  -l, --log                 Output a log file containing STDOUT/ERR from each test and commands
```

Comment:
I find this moniker to generic for the usecase. How about --trace-log?

Reply:
Hey! Thanks for the replies! Isn't `--trace` a different type of output? I tried the code in that PR thinking that it was what I wanted but it seems to output what bats is doing, not command STDOUT/ERR.

What about `--std-log`?

Reply:
Or maybe `--command-std-log`

Reply:
The current trace PR is pretty old and I don't have looked too far into it. My current vision for trace feature would be to have several verbosity levels which add:

1. only commands
2. the outputs

Furthermore, I am hoping for a finegrained control via variables that allows to turn tracing on/off for specific section to avoid overloading the user with output.

The output from commands under `run` is not visible by default but stored in `$output`. Nevertheless, you should be able to print from there if you want it. Then the next hurdle is that output from passing tests is also not printed. You could use FD3 for that, but that has its problems too. I think if we want to have a trace/logging feature, we should also allow for verbosity that enables output even for passing tests.

Reply:
That's awesome, cause I found this too by working on docker-mailserver ;P

I'll take a look at what would be necessary to accomplish what you're thinking. I like the idea. 

What do you want to do with the older trace PR? It looks dead IMO.

Reply:
I meant to revive it but it will definitely need some rework to fit into the current framework. I am afraid that this will still take some time. I am currently working on the last blocker for the 1.4.0 release and won't take more code PRs into that. But I am open to putting this on the track for the 1.5.0 release, which should be due in September.

Reply:
No worries. Would you mind if I work on the trace revival? I'd like to add the verbosity levels you spoke about.

Reply:
Go for it, I am glad for as many helping hands as I can get :smile: , but be warned that it can take some time (just look at the backlog of PRs).

Reply:
@martin-schulze-vireso , any major changes in 1.4.0 that I need to consider before moving forward? Or is it fairly the same structure?

Reply:
I think looking at [this](https://github.com/bats-core/bats-core/blob/master/docs/CHANGELOG.md#unreleased) should give you an idea.

Reply:
There is another PR lined up that changes `run`'s internals which I'd also like to get into the next release and which I'd give higher priority.